lib/machine-loader.js

```diff
@@ -436,6 +477,44 @@ function getNetworkHeartbeatByDevice (deviceId) {
     .then(res => _.mapKeys(_.camelCase, _.find(it => it.device_id === deviceId, res)))
 }
 
+function updateDiagnostics (deviceId, images) {
+  const sql = `UPDATE devices 
+        SET diagnostics_timestamp = NOW(),
+            diagnostics_scan_updated_at = CASE WHEN $2 THEN NOW() ELSE diagnostics_scan_updated_at END,
+            diagnostics_front_updated_at = CASE WHEN $3 THEN NOW() ELSE diagnostics_front_updated_at END
+        WHERE device_id = $1`
+
+  const directory = `${OPERATOR_DATA_DIR}/diagnostics/${deviceId}/`
+  const { scan, front } = images
+
+  return updatePhotos(scan, front, directory)
+    .then(() => db.none(sql, [deviceId, !!scan, !!front]))
+    .catch(err => logger.error('while running machine diagnostics: ', err))
+}
+
+function createPhoto (name, data, dir) {
+  if (!data) {
+    logger.error(`Diagnostics error: No data to save for ${name} photo`)
+    return Promise.resolve()
+  }
+
+  const decodedImageData = Buffer.from(data, 'base64')
+  const filename = path.join(dir, name)
+  return fsPromises.writeFile(filename, decodedImageData)
+}
+
+function updatePhotos (scan, front, dir) {
+  const dirname = path.join(dir)
+  _.attempt(() => makeDir.sync(dirname))
```

Comment:
`_.attempt()` will ignore the error if `makeDir` fails, which will then cause `writeFile()` in `createPhoto` to fail too.